spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -148,6 +148,10 @@ object DistriOptimizer {
     logger.info("Count dataset")
     val countBefore = System.nanoTime()
     val numSamples = dataset.data(train = false).map(_.size()).reduce(_ + _)
+    // FIXME dnn model must cache val dataset first, otherwise there will be a segment fault.
+    if (Engine.getEngineType() == MklDnn) {
+      validationDataSet.get.toDistributed().data(train = false).map(_.size()).reduce(_ + _)
```

Comment:
what if validationDataSet is None?

Reply:
And why do we need this if you already cached validationDataSet in prepareInput?

Reply:
Yes. We should not include this.